src/Features/CSharp/Portable/Snippets/CSharpConstructorSnippetProvider.cs

```diff
@@ -33,13 +46,41 @@ protected override async Task<bool> IsValidSnippetLocationAsync(Document documen
             var semanticModel = await document.ReuseExistingSpeculativeModelAsync(position, cancellationToken).ConfigureAwait(false);
             var syntaxContext = (CSharpSyntaxContext)document.GetRequiredLanguageService<ISyntaxContextService>().CreateContext(document, semanticModel, position, cancellationToken);
 
+            var precedingModifiers = syntaxContext.PrecedingModifiers;
```

Comment:
📝 This is going to conflict with a fairly significant rewrite I just completed in this space. I'm thinking we maybe wait until after that merges.

Reply:
Well, I couldn't predic that. Please ping me here when your changes are merged. For now I'll make this PR a draft

Reply:
If I don't reply back by tomorrow, feel free to ping me or make this ready for review again.